synapse/handlers/space_summary.py

```diff
@@ -446,8 +446,30 @@ async def _is_room_accessible(
 
         room_version = await self._store.get_room_version(room_id)
 
-        # if we have an authenticated requesting user, first check if they are able to view
-        # stripped state in the room.
+        # Include the room if it has join rules of public or knock.
+        join_rules_event_id = state_ids.get((EventTypes.JoinRules, ""))
+        if join_rules_event_id:
+            join_rules_event = await self._store.get_event(join_rules_event_id)
+            join_rule = join_rules_event.content.get("join_rule")
+            if (
+                join_rule == JoinRules.PUBLIC
+                or room_version.msc2403_knocking
+                and join_rule == JoinRules.KNOCK
```

Comment:
```suggestion
                or (room_version.msc2403_knocking
                and join_rule == JoinRules.KNOCK)
```

otherwise I start worrying about operator precedence, which is something I'd rather not think about :slightly_smiling_face: